ui/app/routes/observability/inferences/$inference_id/route.tsx

```diff
@@ -263,6 +264,25 @@ export default function InferencePage({ loaderData }: Route.ComponentProps) {
     }
   }, [addToDatasetFetcher.data, addToDatasetFetcher.state, navigate]);
 
+  const demonstrationFeedbackFetcher = useFetcher<typeof action>();
+  const demonstrationFeedbackFormError =
+    demonstrationFeedbackFetcher.state === "idle"
+      ? (demonstrationFeedbackFetcher.data?.error ?? null)
+      : null;
+  useEffect(() => {
+    const currentState = demonstrationFeedbackFetcher.state;
+    const data = demonstrationFeedbackFetcher.data;
+    if (currentState === "idle" && data?.redirectTo) {
+      navigate(data.redirectTo);
+      setOpenModal(null);
```

Comment:
You shouldn't need to set state here since you're navigating away, right? I think this would create a flash of the new state before the view changes, but either way it seems unnecessary.

Reply:
I tested this and I think the component doesn't unmount on the navigate so the state isn't reset. The navigate is to the same page with new query params so I guess react router is smart enough to not do anything here?